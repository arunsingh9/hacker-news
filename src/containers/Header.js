import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.header`
	background: #ff6600;
`;

const BrandIcon = styled.a`
	color: #fff;
	border: 1px solid #fff;
	padding: 3px 5px;
	display: inline-block;
	margin: 5px;
	text-decoration: none;
	font-size: 0.9rem;
`;

const SkipLink = styled.a`
	color: #1f1414;
	margin: 0 20px;
	padding: 4px;
	font-size: 12px;
`;

export const SkipLinkElement = () => <SkipLink href='#main'>New</SkipLink>;

export const Header = () => {
	return (
		<Wrapper>
			<BrandIcon href='#'>Y</BrandIcon>
			<SkipLinkElement />
		</Wrapper>
	);
};
