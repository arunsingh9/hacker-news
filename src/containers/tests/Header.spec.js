import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { Header, SkipLinkElement } from '../Header';

describe('Header Component', () => {
	test('snapshot renders', () => {
		const component = renderer.create(<Header />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('renders SkipLink ', () => {
		const wrapper = mount(<Header />);
		expect(wrapper.find(SkipLinkElement).length).toEqual(1);
	});
});
