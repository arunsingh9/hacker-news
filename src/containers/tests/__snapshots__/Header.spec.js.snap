// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component snapshot renders 1`] = `
<header
  className="Header__Wrapper-ctQkEG jPUWZB"
>
  <a
    className="Header__BrandIcon-beZnkd coImAA"
    href="#"
  >
    Y
  </a>
  <a
    className="Header__SkipLink-efkLJE eCvdvP"
    href="#main"
  >
    New
  </a>
</header>
`;
